src/Moat.hs

```diff
@@ -729,13 +741,24 @@ consToMoatType o@Options {..} parentName parentDoc instTys variant ts bs = \case
             _ -> do
               mkProd o parentName parentDoc instTys ts con
         _ -> do
-          -- omit the cases we don't want
-          let cons' = flip filter cons $ \ConstructorInfo {..} -> omitCases (nameStr constructorName) == Keep
-          cases <- forM cons' (mkCase o)
-          ourMatch <-
-            matchProxy
-              =<< lift (enumExp parentName parentDoc instTys dataInterfaces dataProtocols dataAnnotations cases dataRawValue ts bs sumOfProductEncodingOptions enumEncodingStyle)
-          pure [pure ourMatch]
+          -- cases which much exist
+          let constructorNames = cons <&> \ConstructorInfo {..} -> nameStr constructorName
+              missingConstructors = strictCases L.\\ constructorNames
+          if null missingConstructors
+            then do
+              -- omit the cases we don't want
```

Comment:
I should probably combine this comment with the one above and make it more clear what this block of code does

Reply:
Agree